packages/issuer/src/blockchain-facade/Certificate.ts

```diff
@@ -0,0 +1,257 @@
+import { TransactionReceipt, EventLog } from 'web3-core';
+
+import { Configuration, BlockchainDataModelEntity, Timestamp } from '@energyweb/utils-general';
+
+import { Registry, PublicIssuer, PUBLIC_CERTIFICATE_TOPIC } from '..';
+
+export interface ICertificate {
+    id: string;
+
+    issuer: string;
+    
+    deviceId: string;
+    generationStartTime: number;
+    generationEndTime: number;
+
+    energy: number;
+    creationTime: number;
+
+    sync(): Promise<ICertificate>;
+
+    getCertificateOwner(): Promise<string>;
+    isClaimed(): Promise<boolean>;
+    claim(amount: number): Promise<TransactionReceipt>;
+
+    transfer(to: string, amount: number): Promise<TransactionReceipt>;
+
+    getAllCertificateEvents(): Promise<EventLog[]>;
+}
+
+const getAccountFromConfiguration = (configuration: Configuration.Entity) => ({
+    from: configuration.blockchainProperties.activeUser.address,
+    privateKey: configuration.blockchainProperties.activeUser.privateKey
+});
+
+export const getAllCertificateEvents = async (
+    certId: number,
+    configuration: Configuration.Entity
+): Promise<EventLog[]> => {
+    const registry: Registry = configuration.blockchainProperties.certificateLogicInstance;
+
+    const allEvents = await registry.getAllEvents(
+        {
+            topics: [
+                null,
+                configuration.blockchainProperties.web3.utils.padLeft(
+                    configuration.blockchainProperties.web3.utils.fromDecimal(certId),
+                    64,
+                    '0'
+                )
+            ],
+            toBlock: undefined
+        }
+    );
+
+    const returnEvents = [];
+
+    for (const fullEvent of allEvents) {
+        // we have to remove some false positives due to ERC721 interface
+        if (fullEvent.event === 'Transfer') {
+            if (fullEvent.returnValues.tokenId === `${certId}`) {
+                returnEvents.push(fullEvent);
+            }
+        } else {
+            returnEvents.push(fullEvent);
+        }
+    }
+
+    // we also have to search
+    if (certId !== 0) {
+        const transferEvents = await registry.getAllTransferSingleEvents(
+            {
+                topics: [
+                    '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef',
+                    null,
+                    null,
+                    configuration.blockchainProperties.web3.utils.padLeft(
+                        configuration.blockchainProperties.web3.utils.fromDecimal(certId),
+                        64,
+                        '0'
+                    )
+                ],
+                toBlock: undefined
```

Comment:
Consider fromBlock: 0/toBlock latest

Reply:
This was copied from old Certificate.ts, will update.